src/include/btree.i

```diff
@@ -981,8 +981,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * been written in the checkpoint, leaving the checkpoint inconsistent.
 	 */
 	if (btree->checkpointing &&
-	    (__wt_page_is_modified(page) ||
-	    F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) {
+	    (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MASK))) {
```

Comment:
It's possible that removing `WT_READ_NO_WAIT` from `__rec_child_modify` may make this unnecessary, but without that change, I don't see anything preventing eviction of a clean page updating its address in the parent (e.g., for `WT_PM_REC_REPLACE`) concurrent with the parent being reconciled during a checkpoint.

In order:

1. if we don't bother maintaining a tree clean/dirty flag, we'll write checkpoints in every open tree on every checkpoint call.  Maybe that's won't cause a measurable performance problem, but I'd want evidence.

2. I think ordinary updates during a checkpoint will trip that assertion, but it's worth a try.

3. `split-insert` only fires for dirty pages, and we prevent eviction of dirty pages during checkpoints.  All of the checks are in `__wt_page_can_evict`.  We can't have pending updates in a clean page regardless: they have to be handled during the initial eviction (before the pages are unlocked) or reads will be wrong.

4. let's measure the performance.  As I said above, removing `NO_WAIT` from `__rec_child_modified` might be sufficient.

Another thought I had here is to have `__rec_child_modified` update the parent page in memory and clear `WT_PM_REC_MASK` on child pages where possible.  That's not simple for splits, though.